miso-web/src/main/webapp/scripts/lims.js

```diff
@@ -1280,7 +1280,8 @@ Utils.validation = {
   dateTimeRegex:
     "^(19|20)[0-9]{2}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01]) ([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9]$",
   sanitizeRegex: "^[^<>&\\t\\n\\r]*$",
-  uriComponentRegex: "^[^<>&%;/\\\\]*$", // problem characters even when encoded in URI components: %;\/
+  uriComponentRegex: "^[^<>&%;/\\\\]*$", // problem characters even when encoded in URI components: <>&%;\/
+  uriRegex: "^[^<>\\\\]*$", // problem characters in full URIs: <>\
```

Comment:
Are new lines and tabs ok in a full uri?

Reply:
A lot more characters are actually bad, but any modern browser will encode them automatically, so it doesn't really matter

Reply:
It also looks like a separate sanitize method at bulk.js:3377 rejects those characters

Reply:
I don't think it's safe to make assumptions about what the client will be using. We should exclude at least \n, \t, \r, and \0

Reply:
We're assuming either way. Preventing characters that must be encoded assumes that the user isn't copying from somewhere that displays URLs unencoded, which is very common since it is widely supported.

I've disallowed tabs and newlines again since they don't really work either way, and it's not a change from the previous behaviour. Further improvements could be made, but the point of this PR is just to allow the ampersand character. If you see value in further improvements, please make a separate ticket.